src/AlgebraicGeometry/Schemes/Divisors/CartierDivisor.jl

```diff
@@ -241,9 +286,7 @@ function irreducible_decomposition(C::EffectiveCartierDivisor)
     components_here = minimal_primes(I_temp)
     for comp in components_here
       I_temp, saturation_index = saturation_with_index(I_temp, comp)
-      temp_dict=IdDict{AbsAffineScheme,Ideal}()
-      temp_dict[U] = comp
-      I_sheaf_temp = IdealSheaf(X, extend!(cov, temp_dict), check=false)
+      I_sheaf_temp = PrimeIdealSheafFromChart(X, U, comp, check=false)
```

Comment:
This is a non-trivial change.

Reply:
I have some doubts about this function `irreducible_decomposition(::EffectiveCartierDivisor)`. 
It seems to assume that this Cartier divisor is a Weyl divisor. Do we need to assume the ambient scheme $X$ to be regular in codimension one for this to work? And if it is not, what is computed here?

Also I wonder if we could return a Weyl divisor here.